x-pack/plugins/observability_ai_assistant/server/service/knowledge_base_service/index.ts

```diff
@@ -241,7 +353,43 @@ export class KnowledgeBaseService {
     }
   };
 
-  summarize = async ({
+  getEntries = async (query: string | undefined): Promise<{ entries: KnowledgeBaseEntry[] }> => {
+    try {
+      const response = await this.dependencies.esClient.search<KnowledgeBaseEntry>({
+        index: this.dependencies.resources.aliases.kb,
+        ...(query
+          ? {
+              query: {
+                wildcard: {
+                  'labels.document_id': {
+                    value: `${query}*`,
```

Comment:
yeah leading wildcards are expensive and sometimes not allowed: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl.html#query-dsl-allow-expensive-queries.

do we only want to search by ID here, and not context? Also, why do we use labels.document_id? does the query not work on `_id`?

Reply:
With context do you mean the full text of the item?

I tried the query of _id but that threw an error. Open to suggestions on what to use instead?

Reply:
Also want to add sorting

Reply:
if _id throws an error, let's add it as a top-level property. Generally labels should be user-defined. (Although I'm not sure if I've stuck to this principle myself 😄 ). Which reminds me, what happens with documents that were indexed before this release? Do we have a backwards compatibility approach?

Reply:
A good point. I'll add some stuff to account for this scenario

Reply:
I don't know how to do this.

Reply:
Basically, these documents need to still show up, or be migrated on Kibana startup. Happy to hop on a call. We can also just make the breaking change given we are not GA yet, but let's try to avoid it.